setup_pulp.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+#
+# This script sets up a pulp all-in-one container (per the instructions at
+# https://pulpproject.org/pulp-in-one-container/) that the builder tests can
+# then use as an image repository that supports image signatures.
+#
+
+set -ex
+
+##############################################################################
+# Add sigstore configurations for the images we wish to mirror.
+##############################################################################
+
+mkdir -p ~/.config/containers/registries.d
+cat <<EOF >> ~/.config/containers/registries.d/registry.access.redhat.com.yaml
+docker:
+     registry.access.redhat.com:
+         sigstore: https://access.redhat.com/webassets/docker/content/sigstore
+EOF
+
+cat <<EOF >> ~/.config/containers/registries.d/registry.redhat.io.yaml
+docker:
+     registry.redhat.io:
+         sigstore: https://registry.redhat.io/containers/sigstore
+EOF
+
+
+##############################################################################
+# Pull and run the pulp container
+##############################################################################
+
+podman pull docker.io/pulp/pulp:3.19
+
+mkdir pulp
+cd pulp
+mkdir settings pulp_storage pgsql containers
+
+echo "CONTENT_ORIGIN='http://$(hostname):8080'
+ANSIBLE_API_HOSTNAME='http://$(hostname):8080'
+ANSIBLE_CONTENT_HOSTNAME='http://$(hostname):8080/pulp/content'
+TOKEN_AUTH_DISABLED=True" >> settings/settings.py
+
+podman run --detach \
+           --publish 8080:80 \
+           --name pulp \
+           --volume "$(pwd)/settings":/etc/pulp \
+           --volume "$(pwd)/pulp_storage":/var/lib/pulp \
+           --volume "$(pwd)/pgsql":/var/lib/pgsql \
+           --volume "$(pwd)/containers":/var/lib/containers \
+           --device /dev/fuse \
+           pulp/pulp
+
+##############################################################################
+# Iteratively query the REST API until we get a JSON response (http code will
+# be 200) which will indicate the system is ready.
+##############################################################################
+
+set +e
+iter=0
+httpcode=$(curl -o /dev/null -s -w "%{http_code}\n" localhost:8080/pulp/api/v3/status/)
+
+until [ "$httpcode" = "200" ]
+do
+  iter=$((iter+1))
+  if [ $iter == 20 ]; then
+    exit 1
```

Comment:
Given core's past experience with the pulp container DB init just going to lunch and never coming back once in awhile, might be good to add an error message if this exit trips.